packages/wrangler/src/sites.tsx

```diff
@@ -166,21 +173,28 @@ export async function syncAssets(
     }
 
     // remove the key from the set so we know what we've already uploaded
-    keys.delete(assetKey);
+    namespaceKeys.delete(assetKey);
     manifest[path.relative(siteAssets.assetDirectory, absAssetFile)] = assetKey;
   }
 
   // keys now contains all the files we're deleting
-  for (const key of keys) {
+  for (const key of namespaceKeys) {
     console.log(`deleting ${key} from the asset store...`);
   }
 
-  await Promise.all([
-    // upload all the new assets
-    putBulkKeyValue(accountId, namespace, toUpload, () => {}),
-    // delete all the unused assets
-    deleteBulkKeyValue(accountId, namespace, Array.from(keys), () => {}),
-  ]);
+  if (!dryRun) {
```

Comment:
Since the code now returns early if on a dry-run, I don't think this change is needed any more.

Reply:
good catch yes!